src/Database/Adapter/Mongo/MongoClient.php

```diff
@@ -0,0 +1,563 @@
+   */
+
+  // https://docs.mongodb.com/manual/reference/command/listCollections/#listcollections
+  public function listCollectionNames($filter = [], $options = []) {
+    $qry = array_merge([
+      "listCollections" => 1.0,
+      "nameOnly" => true,
+      "authorizedCollections" => true,
+      "filter" => $this->toObject($filter)],
+      $options
+    );
+
+    return $this->query($qry);
+  }
+
+  /**
+   * Create indexes.
+   * 
+   * @param string $collection
+   * @param array $indexes
+   * @param array $options
+   * 
+   * @return array
+   */
+
+  // https://docs.mongodb.com/manual/reference/command/createIndexes/#createindexes
+  public function createIndexes($collection, $indexes, $options = []) {
+
+    $qry = array_merge(
+      [
+        'createIndexes' => $collection, 
+        'indexes' => $indexes
+      ], $options);
+
+    $this->query($qry);
+
+    return true;
+  }
+
+  /**
+   * Drop indexes from a collection.
+   * 
+   * @param string $collection
+   * @param array $indexes
+   * @param array $options
+   * 
+   * @return array
+   */
+  // https://docs.mongodb.com/manual/reference/command/dropIndexes/#dropindexes
+  public function dropIndexes(string $collection, $indexes, $options = []) {
+    $this->query(array_merge([
+        'dropIndexes' => $collection,
+        'index' => $indexes,
+      ], $options)
+    );
+
+    return $this;
+  }
+
+  /**
+   * Insert a document/s.
+   * 
+   * @param string $collection
+   * @param array $documents
+   * @param array $options
+   * 
+   * @return Document
+   */
+
+  // https://docs.mongodb.com/manual/reference/command/insert/#mongodb-dbcommand-dbcmd.insert
+  public function insert(string $collection, $documents, $options = []) {
+    $docObj = new \stdClass();
+
+    foreach($documents as $key => $value) {
+        $docObj->{$key} = $value;
+    }
+
+    $result = $this->query(array_merge([
+      MongoCommand::INSERT => $collection, 
+      'documents' => [$docObj],
+    ], $options));
+
+    return new Document($this->toArray($docObj));
+  }
+
+  /**
+   * Update a document/s.
+   * 
+   * @param string $collection
+   * @param array $documents
+   * @param array $options
+   * 
+   * @return MongoClient
+   */
+
+  // https://docs.mongodb.com/manual/reference/command/update/#syntax
+  public function update(string $collection, $where = [], $updates = [], $options = []) {
+
+    $this->query(array_merge([
+      MongoCommand::UPDATE => $collection, 
+      'updates' => [
+          [
+            "q" => $this->toObject($where),
+            "u" => $this->toObject($updates),
+            'multi' => false,
+            'upsert' => false
+          ]
+        ]
+      ], $options)
+    );
+
+    return $this;
+  }
+
+  /**
+   * Insert, or update, a document/s.
+   * 
+   * @param string $collection
+   * @param array $documents
+   * @param array $options
+   * 
+   * @return MongoClient
+   */
+
+
+  // https://docs.mongodb.com/manual/reference/command/update/#syntax
```

Comment:
```suggestion
  /**
   * Insert, or update, a document/s.
   * 
   * https://docs.mongodb.com/manual/reference/command/update/#syntax
   * @param string $collection
   * @param array $documents
   * @param array $options
   * 
   * @return MongoClient
   */
```

I think this looks a lot cleaner 🤔